SnailTests/ObservableTests.swift

```diff
@@ -1054,4 +1054,63 @@ class ObservableTests: XCTestCase {
 
         subject.value = true
     }
+
+    func testRemovingFirstSubscriberWithinSubscription_NoDeadlock() {
```

Comment:
Running this test on current main causes the tests to hang (deadlock). Afaik there isn't a way to get a test to fail if deadlock occurs.